src/BaseDrawableMesh.ts

```diff
@@ -0,0 +1,109 @@
+import { Euler, Group, Material, Mesh, Vector3 } from "three";
+import { IDrawableObject } from "./types";
+
+/**
+ * Abstract base class for drawable 3D mesh objects.
+ *
+ * Provides default implementations for some methods in the IDrawableObject
+ * interface, including handling for visibility, transformations, and cleanup.
+ *
+ * As a default, subclasses should call `this.addChildMesh(mesh)` to register
+ * any meshes that should be managed by the base class.
+ */
+export default abstract class BaseDrawableMesh implements IDrawableObject {
+  /**
+   * Pivot group that all child meshes are parented to. Transformations are
+   * applied to this group.
+   */
+  protected readonly meshPivot: Group;
+  protected scale: Vector3;
+  protected flipAxes: Vector3;
+
+  /**
+   * Child meshes that are part of this drawable object.
+   */
+  protected readonly meshes: Set<Mesh>;
+
+  constructor() {
+    this.meshPivot = new Group();
+    this.scale = new Vector3(1, 1, 1);
+    this.flipAxes = new Vector3(1, 1, 1);
+
+    this.meshes = new Set<Mesh>();
+  }
+
+  protected addChildMesh(mesh: Mesh): void {
+    if (!this.meshes.has(mesh)) {
```

Comment:
I wonder if this should be managed better from outside. Or maybe throw some kind of error if a mesh got added twice. How likely could it happen?